test/test-clusters/src/main/java/org/elasticsearch/test/cluster/util/Pair.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.cluster.util;
+
+import java.util.Objects;
+
+public final class Pair<L, R> {
```

Comment:
Instead of creating yet another tuple class, can we just create a record with concrete types as necessary?

Reply:
We could, but I specifically avoided using records so this can be backported to 7.17 more easily.